internal/engine/synclet_build_and_deployer.go

```diff
@@ -89,8 +118,13 @@ func (sbd *SyncletBuildAndDeployer) canSyncletBuild(ctx context.Context,
 	}
 
 	// Can't do container update if we don't know what container manifest is running in.
-	if _, ok := sbd.getDeployInfoForImage(state.LastResult.Image); !ok {
-		return fmt.Errorf("no container info for this manifest")
+	info, ok := sbd.deployInfoForImageBlocking(ctx, state.LastResult.Image)
+	if !ok || info == nil {
+		return fmt.Errorf("have not yet fetched deploy for this manifest")
```

Comment:
This seems like a weird error to return. What do you think about populating and blocking instead of erroring here (or probably inside `deployInfoForImageBlocking`)?

Reply:
thaaaat seems reasonable -- will leave a todo

Reply:
legit -- don't wanna gum up this PR too much but i left a todo / [made a ticket](https://app.clubhouse.io/windmill/story/376/if-we-try-to-build-via-synclet-and-don-t-have-deployinfo-fetch-it)